### DISCUSSION TITLE
Patch update process

### DISCUSSION THREAD

**Discussion by farfromrefug** [VSCodium/vscodium#1053] [Q&A] [answered]
I am trying to build 1.67 nightly because 1.66 have an known CPU usage issue which should be fixed in 1.67. And my computer is burning out right now ...
Now the build fails because one patch is not up to date anymore. I am ok with fixing it and creating a PR but i dont know what the process is to update a patch ?
Could someone guide me? We could even add it to the readme, it could help contributions

**Comment by daiyam** [comment, marked as answer]
@farfromrefug Hi, here are my steps:
- run `./build/build_linux.sh`, if a patch is failing then,
- open `vscode` directory in vscodium
- revert all changes
- run `git apply --reject ../patches/use-github-pat.patch`
- fix `*.rej`
- run `yarn watch`
- run `./script/code.sh` until everything ok
- run `git diff > ../patches/use-github-pat.patch`

**Reply by farfromrefug** [reply]
@daiyam thanks that s what i was looking for! Now i am stuck with vscode which wont work on my machine. I got errors all over the place. Will try and get through them.

**Reply by farfromrefug** [reply]
@daiyam thanks for the process i can confirm it works. Wondering if there could be an easier way. Like maintaining a fork of vscode with the changes from vscodium. That way merge would work and we could simply use that fork to easily update patches. We could even create patches in a github action automatically (except for when merge fails).
Git diff allows to create diff patches between branches. We could do it with vscodium changes vs master: 
`git diff master..vscodium`

**Reply by farfromrefug** [reply]
@daiyam thanks to you i successfully released the insiders build for macos from a github action! https://github.com/Akylas/vscodium/releases/tag/1.67.0-insider
If you are interested i had to add 2 fixes:
* build fix https://github.com/Akylas/vscodium/commit/369ae7fc0a375eef7bfac73beca9933488d9cc37
* patch update for 1.67 https://github.com/Akylas/vscodium/commit/6ab744df13734b454c9873a8dcd2fd3c015610a8

**Reply by daiyam** [reply]
I was thinking to release some vscodium-insiders <del>but I'm afraid we might run into the quota of the actions runtime</del>.
I will need to finish #960...

Edit: I'm stupid: "Minutes are free for public repositories." I've just had to look at the little lines...

**Reply by farfromrefug** [reply]
@daiyam you are right i actually disable most workflow triggers for my use case.
And what about a weekly insider for a start? it is like 4 times faster that vscode release process. Or maybe twice a week

EDIT: why do you have the workflows running on every commit? https://github.com/VSCodium/vscodium/blob/master/.github/workflows/macos.yml#L7= (if read it correctly)

**Reply by daiyam** [reply]
@farfromrefug No quota for public repositories. cf my edit...

Anyway, I was thinking about a weekly release.

**Reply by paulcarroty** [reply]
Weekly seems fine. Maybe worth to add the parching guide^ to insiders README.

**Reply by daiyam** [reply]
I've started a new discussion about the Insiders version at #1057.
